tensorflow/c/kernels.cc

```diff
@@ -279,4 +279,33 @@ TF_Tensor* TF_AllocateOutput(TF_OpKernelContext* context, int index,
     return nullptr;
   }
   return tf_tensor;
-}
\ No newline at end of file
+}
+
+TF_Tensor* TF_AllocateTemp(TF_OpKernelContext* context, TF_DataType dtype, 
+                           int64_t* dims, int num_dims, 
+                           TF_AllocatorAttributes* attributes,
+                           TF_Status* status) {
+  auto* cc_ctx = reinterpret_cast<::tensorflow::OpKernelContext*>(context);
+  TF_SetStatus(status, TF_OK, ""); 
+  tensorflow::gtl::ArraySlice<tensorflow::int64> dimarray(
+      reinterpret_cast<tensorflow::int64*>(dims), num_dims);
+  tensorflow::AllocatorAttributes allocator_attr; 
+  if (attributes->on_host) { 
```

Comment:
Comment from @yisitu : Please add input validation:
```cpp
if (!attributes || !attributes->struct_size) {
  // If status is default TF_OK, set it to error here.
  return nullptr;
}
```

My own comment:
I guess we could allow `attributes` to be nullptr and in that case create blank `AllocatorAttributes`. But we should definitely check for size:
TF_SetStatus(status, TF_INVALID_ARGUMENT, "TF_AllocatorAttributes struct_size member must be set to TF_ALLOCATOR_ATTRIBUTES_STRUCT_SIZE");